File: db4on/reference/db4o-reference-chapters/src/com/db4odoc/f1/utility/UtilityExample.cs
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */
using System;
using System.IO;
using com.db4o;


namespace com.db4odoc.f1.utility
{
	public class UtilityExample
	{
		public static void main(String[] args) 
		{
			TestDescend();
			CheckActive();
			CheckStored();
		}

		public static void StoreSensorPanel()
		{
			File.Delete(Util.YapFileName);
			ObjectContainer db = Db4o.OpenFile(Util.YapFileName);
			try 
			{
				// create a linked list with length 10
				SensorPanel list = new SensorPanel().CreateList(10); 
				// store all elements with one statement, since all elements are new		
				db.Set(list);
			} 
			finally 
			{
				db.Close();
			}
		}
	
		public static void TestDescend()
		{
			StoreSensorPanel();
			ObjectContainer db = Db4o.OpenFile(Util.YapFileName);
			try 
			{
				db.Ext().Configure().ActivationDepth(1);
				System.Console.WriteLine("Object container activation depth = 1");
				ObjectSet result = db.Get(new SensorPanel(1));
				SensorPanel spParent = (SensorPanel)result[0];
				SensorPanel spDescend = (SensorPanel)db.Ext().Descend((Object)spParent, new String[]{"_next","_next","_next","_next","_next"});
				db.Ext().Activate(spDescend, 5);
				System.Console.WriteLine(spDescend);
			} 
			finally 
			{
				db.Close();
			}
		}
	
		public static void CheckActive()
		{
			StoreSensorPanel();
			ObjectContainer db = Db4o.OpenFile(Util.YapFileName);
			try 
			{
				db.Ext().Configure().ActivationDepth(2);
				System.Console.WriteLine("Object container activation depth = 2");
				ObjectSet result = db.Get(new SensorPanel(1));
				SensorPanel sensor = (SensorPanel)result[0]; 
				SensorPanel next = sensor.Next;
				while (next != null)
				{
					System.Console.WriteLine("Object " + next +" is active: " + db.Ext().IsActive(next));
					next = next.Next;
				}
			} 
			finally 
			{
				db.Close();
			}
		}
	
		public static void CheckStored()
		{
			// create a linked list with length 10
			SensorPanel list = new SensorPanel().CreateList(10);
			File.Delete(Util.YapFileName);
			ObjectContainer db = Db4o.OpenFile(Util.YapFileName);
			try 
			{
				// store all elements with one statement, since all elements are new		
				db.Set(list);
				Object sensor = (Object)list.Sensor;
				SensorPanel sp5 = list.Next.Next.Next.Next;
				System.Console.WriteLine("Root element "+list+" isStored: " + db.Ext().IsStored(list));
				System.Console.WriteLine("Simple type  "+sensor+" isStored: " + db.Ext().IsStored(sensor));
				System.Console.WriteLine("Descend element  "+sp5+" isStored: " + db.Ext().IsStored(sp5));
				db.Delete(list);
				System.Console.WriteLine("Root element "+list+" isStored: " + db.Ext().IsStored(list));
			} 
			finally 
			{
				db.Close();
			}
		}
	}
}

File: db4on/reference/vb/db4o-reference-chapters/src/com/db4odoc/f1/utility/UtilityExample.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 
Imports System
Imports System.IO
Imports com.db4o


Namespace com.db4odoc.f1.utility
    Public Class UtilityExample
        Public Shared ReadOnly YapFileName As String = "formula1.yap"

        Public Shared Sub Main(ByVal args() As String)
            TestDescend()
            CheckActive()
            CheckStored()
        End Sub
        ' end Main

        Public Shared Sub StoreSensorPanel()
            File.Delete(YapFileName)
            Dim db As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                ' create a linked list with length 10
                Dim list As SensorPanel = New SensorPanel().CreateList(10)
                ' store all elements with one statement, since all elements are new		
                db.Set(list)
            Finally
                db.Close()
            End Try
        End Sub
        ' end StoreSensorPanel

        Public Shared Sub TestDescend()
            StoreSensorPanel()
            Dim db As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                db.Ext().Configure().ActivationDepth(1)
                System.Console.WriteLine("Object container activation depth = 1")
                Dim result As ObjectSet = db.Get(New SensorPanel(1))
                Dim spParent As SensorPanel = CType(result(0), SensorPanel)
                Dim fields() As String = {"_next", "_next", "_next", "_next", "_next"}
                Dim spDescend As SensorPanel = CType(db.Ext().Descend(CType(spParent, Object), fields), Object)
                db.Ext().Activate(spDescend, 5)
                System.Console.WriteLine(spDescend)
            Finally
                db.Close()
            End Try
        End Sub
        ' end TestDescend

        Public Shared Sub CheckActive()
            StoreSensorPanel()
            Dim db As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                db.Ext().Configure().ActivationDepth(2)
                System.Console.WriteLine("Object container activation depth = 2")
                Dim result As ObjectSet = db.Get(New SensorPanel(1))
                Dim sensor As SensorPanel = CType(result(0), SensorPanel)
                Dim NextSensor As SensorPanel = sensor.NextSensor
                While Not NextSensor Is Nothing
                    System.Console.WriteLine("Object " + NextSensor.ToString() + " is active: " + db.Ext().IsActive(NextSensor).ToString())
                    NextSensor = NextSensor.NextSensor
                End While
            Finally
                db.Close()
            End Try
        End Sub
        ' end CheckActive

        Public Shared Sub CheckStored()
            ' create a linked list with length 10
            Dim list As SensorPanel = New SensorPanel().CreateList(10)
            File.Delete(YapFileName)
            Dim db As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                ' store all elements with one statement, since all elements are new		
                db.Set(list)
                Dim sensor As Object = CType(list.Sensor, Object)
                Dim sp5 As SensorPanel = list.NextSensor.NextSensor.NextSensor.NextSensor
                System.Console.WriteLine("Root element " + list.ToString() + " isStored: " + db.Ext().IsStored(list).ToString())
                System.Console.WriteLine("Simple type  " + sensor.ToString() + " isStored: " + db.Ext().IsStored(sensor).ToString())
                System.Console.WriteLine("Descend element  " + sp5.ToString() + " isStored: " + db.Ext().IsStored(sp5).ToString())
                db.Delete(list)
                System.Console.WriteLine("Root element " + list.ToString() + " isStored: " + db.Ext().IsStored(list).ToString())
            Finally
                db.Close()
            End Try
        End Sub
        ' end CheckStored
    End Class
End Namespace
